scripts/on_sd_start.sh

```diff
@@ -63,11 +63,25 @@ case "${OS_NAME}" in
     *)          echo "Unknown OS: $OS_NAME! This script runs only on Linux or Mac" && exit
 esac
 
+# This is the "main" GPU, but some systems will have extra GPUs as "Display Controllers"
+# Notably, AMD laptops with a AMD dGPU, but also probably any desktop with a second GPU.
+# For nvidia it already works fine
+GPU_BRAND=$(lspci 2>/dev/null | grep VGA)
+DGPU_BRAND=$(lspci 2>/dev/null | grep Display)
+
 # install torch and torchvision
 if python ../scripts/check_modules.py torch torchvision; then
     # temp fix for installations that installed torch 2.0 by mistake
     if [ "$OS_NAME" == "linux" ]; then
-        python -m pip install --upgrade torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116 -q
+        # Check for AMD dGPU first, assume nobody has a primary nvidia GPU and AMD secondary GPU,
+        # and if they do they can edit this themselves like AMD users already have to
+        if echo "$DGPU_BRAND" | grep -q "AMD"; then
+            python -m pip install --upgrade torch torchvision --extra-index-url "https://download.pytorch.org/whl/rocm5.4.2" -q
+        elif echo "$GPU_BRAND" | grep -q "AMD"; then
+            python -m pip install --upgrade torch torchvision --extra-index-url "https://download.pytorch.org/whl/rocm5.4.2" -q
+        else
+            python -m pip install --upgrade torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116 -q
+        fi
```

Comment:
Multi-GPU setups will get the (slower) ROCm installed, even if a more powerfull NVidia card would also be present.

Reply:
Would detecting if theres an AMD *or* NVIDIA dGPU, and always prioritizing an NVIDIA one, be acceptable?

Reply:
Yes, I think prioritizing an NVIDIA GPU (over AMD) is a good compromise for now. We can revisit this in the future if it becomes an issue.